snakemake/dag.py

```diff
@@ -996,7 +996,9 @@ def update_needrun(job):
                 output_mintime_ = output_mintime.get(job)
                 if output_mintime_:
                     updated_input = [
-                        f for f in job.input if f.exists and f.is_newer(output_mintime_)
+                        f
+                        for f in job.input
+                        if (f.exists and f.is_newer(output_mintime_)) or (not f.exists)
```

Comment:
@johanneskoester

In order to handle new missing files returned by an input function (after a first successful workflow run) I tried to change the condition by adding `or (not f.exists)` but this is not a valid option because of temporary output.

I'm currently trying to replace the condition (not yet pushed) `if (f.exists and f.is_newer(output_mintime_))` by something using a new flag like "missing": `if (f.exists and f.is_newer(output_mintime_)) or (not f.exists and is_flagged(f, "missing")`.

So I added the new flag "missing" in the code but I discovered that flags are not "forwarded" to input function elements during wildcards resolution.  I don't know how to do that and if you think that it would be a good idea.

Reply:
I've just pushed a new version where the new "missing" flag is applied correctly to all element yield by the input function.